pyqtorch/utils.py

```diff
@@ -414,6 +413,27 @@ def promote_operator(operator: Tensor, target: int, n_qubits: int) -> Tensor:
     return operator
 
 
+def permute_basis(operator: Tensor, qubit_support: tuple) -> Tensor:
+    """Takes an operator tensor and permutes the rows and
+    columns according to the order of the qubit support.
+
+    Args:
+        operator (Tensor): Operator to permute over.
+        qubit_support (tuple): Qubit support.
+
+    Returns:
+        Tensor: Permuted operator.
+    """
+    n_qubits = len(qubit_support)
+    batch_size = operator.size(-1)
+    operator = operator.view([2] * 2 * n_qubits + [batch_size])
+    ordered_support = argsort(qubit_support)
+    perm = list(
+        tuple(ordered_support) + tuple(ordered_support + n_qubits) + (2 * n_qubits,)
+    )
+    return operator.permute(perm).reshape([2**n_qubits, 2**n_qubits, batch_size])
```

Comment:
Can you add a check that if `qubit_support == tuple(sorted(qubit_support))` it does early return of the operator? That way we for sure avoid calling the `permute` and `reshape` if not needed.

Reply:
Possibly those functions are also smart not to do anything if they don't need to.